File: backend spring/src/main/java/com/bloodbank/entity/DonationRecord.java
package com.bloodbank.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "donation_records")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DonationRecord {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "record_id")
    private Long recordId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "donor_user_id", nullable = false)
    private User donor;

    @Column(name = "donation_date", nullable = false)
    private LocalDate donationDate;

    @Column(name = "units_donated", nullable = false)
    private int unitsDonated = 1; // Default to 1 unit per donation

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;
}

File: BackendDotNet/Models/DonationRecord.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackendDotNet.Models
{
    [Table("donation_records")]
    public class DonationRecord
    {
        [Key]
        [Column("record_id")]
        public long RecordId { get; set; }

        [Required]
        [Column("donor_user_id")]
        public long DonorUserId { get; set; }

        [Required]
        [Column("donation_date")]
        public DateTime DonationDate { get; set; }

        [Required]
        [Column("units_donated")]
        public int UnitsDonated { get; set; } = 1;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation property
        [ForeignKey("DonorUserId")]
        public virtual User Donor { get; set; } = null!;
    }
}
